src/Hint/Context.hs

```diff
@@ -197,18 +197,25 @@ isPhantomModule mn = do (as,zs) <- getPhantomModules
 loadModules :: MonadInterpreter m => [String] -> m ()
 loadModules fs = do -- first, unload everything, and do some clean-up
                     reset
-                    doLoad fs `catchIE` (\e -> reset >> throwM e)
-
-doLoad :: MonadInterpreter m => [String] -> m ()
-doLoad fs = mayFail $ do
-                   targets <- mapM (\f->runGhc2 GHC.guessTarget f Nothing) fs
-                   --
-                   runGhc1 GHC.setTargets targets
-                   res <- runGhc1 GHC.load GHC.LoadAllTargets
+                    targets <- mapM (\f->runGhc2 GHC.guessTarget f Nothing) fs
+                    runGhc1 GHC.setTargets targets
+                    catchReset (doLoad GHC.LoadAllTargets)
```

Comment:
I meant that this line could just call `reload`, unless I'm missing something. Then, `catchReset` would be unnecessary.